crates/driver/src/infra/tokens.rs

```diff
@@ -140,30 +149,43 @@ impl Inner {
         futures::future::join_all(futures).await
     }
 
+    /// Ensures that all the missing tokens are in the cache afterwards while
+    /// taking into account that the function might be called multiple times
+    /// for the same tokens.
```

Comment:
This is a pretty strong assumption (which may not be true e.g. for concurrent quote requests). Would it be simple/performant enough to fetch token info one by one (should be batched on transport level) and use shared requests?

Reply:
I did not try out `RequestSharing` because I was worried the implementation would end up more complex than what I had.
But it turns out that as readable IMO + it should have a higher throughput when different drivers are fetching metadata for different tokens in parallel.
The latency for the case where we fetch Metadata for all tokens in an auction didn't really change with `RequestSharing` btw.